exec/java-exec/src/main/java/org/apache/drill/exec/rpc/security/ClientAuthenticatorProvider.java

```diff
@@ -57,17 +57,17 @@ private ClientAuthenticatorProvider() {
 
     // then, custom factories
     if (customFactories != null) {
-      try {
-        final String[] factories = customFactories.split(",");
-        for (final String factory : factories) {
+      final String[] factories = customFactories.split(",");
+      for (final String factory : factories) {
+        try {
           final Class<?> clazz = Class.forName(factory);
           if (AuthenticatorFactory.class.isAssignableFrom(clazz)) {
             final AuthenticatorFactory instance = (AuthenticatorFactory) clazz.newInstance();
             authFactories.put(instance.getSimpleName(), instance);
           }
+        } catch (final ClassNotFoundException | IllegalAccessException | InstantiationException e) {
```

Comment:
catch ReflectiveOperationException.

Reply:
a moot point, I think that three specific exceptions is better the one general.